.github/workflows/release-publish.yml

```diff
@@ -0,0 +1,158 @@
+    # and provides the output `VERSION`
+    - name: Get the tag name
+      id: get_version
+      run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+
+    # Log changelog (FYI only)
+    - name: Changelog for ${{ steps.get_version.outputs.VERSION }}
+      env:
+        RELEASE_VERSION: ${{ steps.get_version.outputs.VERSION }}
+      run: |
+        git changelog -t ${RELEASE_VERSION} -n -x /dev/null
+
+    # Deploys Maven artifacts. Build and test steps were already ran in previous steps.
+    # Not running tests, because the environment contains secrets.
+    - name: Publish Maven artifacts for release ${{ steps.get_version.outputs.VERSION }}
+      env:
+        MAVEN_USERNAME: ${{ secrets.OSSRH_ACCESS_ID }}
+        MAVEN_OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
+        MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+        SPARK_LOCAL_IP: localhost
+      run: mvn -B deploy --file pom.xml -Pnative -Prelease -DdeployAtEnd=true -DskipTests
+
+    # Deploys Gradle plugin. Build and test steps were already ran in previous steps
+    # TODO this needs the Gradle-plugin-portal secrets
+    - name: Publish Gradle Plugin for release ${{ steps.get_version.outputs.VERSION }}
+      working-directory: ./tools/apprunner-gradle-plugin/
+      run: ./gradlew publishPlugins
+
+    # Deploys pynessie. Build and test steps were already ran in previous steps
+    - name: Build pynessie for release ${{ steps.get_version.outputs.VERSION }}
+      working-directory: ./python
+      run: python setup.py sdist bdist_wheel
+
+    - name: Publish pynessie package
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN }}
+
+    # Deploys Docker images. Build and test steps were already ran in previous workflows
+    # TODO this needs the pypi secrets
+    - name: Publish Docker image for release ${{ steps.get_version.outputs.VERSION }}
+      env:
+        RELEASE_VERSION: ${{ steps.get_version.outputs.VERSION }}
+      run: |
+        echo '${{ secrets.DOCKER_TOKEN }}' | docker login -u '${{ secrets.DOCKER_USERNAME }}' --password-stdin
+        docker tag projectnessie/nessie:${RELEASE_VERSION} projectnessie/nessie:latest
+        docker push projectnessie/nessie:${RELEASE_VERSION} && docker push projectnessie/nessie:latest
+
+    # TODO close + deploy the OSSRH staging repo, maybe switch from maven-deploy-plugin to nexus-staging-maven-plugin to allow that?
+
+    # Create Nessie release on GitHub
+    # TODO need some templating to create a release-notes markdown with highlights, changelog and download locations.
+    - name: Create Nessie release ${{ steps.get_version.outputs.VERSION }} in GitHub
+      env:
+        RELEASE_VERSION: ${{ steps.get_version.outputs.VERSION }}
+      run: |
+        git changelog -t ${RELEASE_VERSION} -n -x /dev/null | egrep -v '^  [*] (Bump |\[release\] ).*$' > /tmp/release-notes.txt
```

Comment:
Is the plan to manually edit the release notes to enrich it or add a simple template?

Reply:
No concrete plan here yet.

Some idea is to use plumb together some blocks to a markdown:
1. Highlights from site/docs/try/releases.md (but no clue how to easily extract it from there in a simple and non-error-prone way)
2. Changelog from `git changelog` (or `git log` with some options)
3. Links, download info, etc

Reply:
cool, so likely manually to start